pkg/assembler/nodes.go

```diff
@@ -15,10 +15,15 @@
 
 package assembler
 
+type Metadata struct {
+	Source string
```

Comment:
Should this mirror source information with both collector and source ? or ?

Reply:
let's add a comment for the Metadata and its fields since its not super straightforward what they mean.

Reply:
Added both source and collector.